### DISCUSSION TITLE
using Prisma inside modules

### DISCUSSION THREAD

**Discussion by cpt-n3mo** [prisma/prisma#14552] [Q&A] [answered]
Good day..

im working on a project that is using prisma.
im just keep running into this very annoying issue that i cant seem to figure out.

my project structure is as followed 

`/src/core/index.js` << module to be used, this module contains prisma.
`/src/testing/app.js` << test application that uses the above module

now when i installing the core module inside the test project. and want to use the module, it keeps complaining about 
`error: @prisma/client did not initialize yet. Please run "prisma generate" and try to import it again.`

this keeps happening what ever i do.. i do NOT want to install prisma separately inside the test project, as it should stay in its module.
while prisma file should be inside the test project.

but i cant seem to be able to get this to work.. it runs fine if i just run the module directly. so running `/src/core/index.js` works without a problem.

but it seems to be impossible to use prisma as a module as it keeps giving the above error..

what is the best way of using prisma in this way. as the docs say absolutely nothing about this what so ever,. to be clear the module is running prisma generate before hand. but still fails with the above error.

kind regards

**Comment by austincrim** [comment, marked as answer]
Hey @cpt-n3mo 👋,

Could you provide a small reproduction of your project setup? That will help a lot as I try to reproduce your error.

**Reply by cpt-n3mo** [reply]
Good day.
thank you for your reply!

is it possible to do that..? as in creating a node module, then using it locally, in a sandbox.?
i did not know that was possible, i will try my best then to make a reproduction..

kind regards.

**Reply by austincrim** [reply]
I might be misunderstanding you, but I'm ideally looking for a GitHub repo with the same folder structure and code that you are having trouble with.

**Reply by cpt-n3mo** [reply]
ah yeh git repo would be easier, but ..  I seem to run in to trouble reproducing the error myself on my home machine.
I sort of fixed it a while back but it just returned "with a fresh clone".. im not sure what is different between the systems, i will have to take a look, bough have a fresh clone, 
in de past i fixed it by  adding the following to my module index.js, its the first thing the module does on init 
`            execSync(`npx prisma generate schema=${join(process.cwd() + '/schema.prisma')}`, { cwd: join(__dirname + "/../"), shell: true, stdio: "inherit" })
`
this forces prisma client to look in the right place for the init file "inside the module's node_modules", and will try to find prisma schema in the process root of the test app using the module.. this feels abit like a hack though. 

not sure why my work machine is not taking this though.. i will have to take a look..

Sorry for posting,  and taking up your time..

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
